apps/nsqd/nsqd.go

```diff
@@ -220,6 +220,13 @@ func (p *program) Start() error {
 	cfg.Validate()
 
 	options.Resolve(opts, flagSet, cfg)
+
+	for _, v := range opts.E2EProcessingLatencyPercentiles {
+		if v <= 0 || v > 1 {
```

Comment:
this line:

https://github.com/bmizerany/perks/blob/master/quantile/stream.go#L62

makes me think that 1.0 might not be an acceptable value ... though I don't understand the algorithm and can't say for sure

Reply:
It seems like 1.0 is legit, since the math you pointed at evaluates to `+Inf` in this case, and thus should not affect the `if f < m` logic a couple of lines below.

Reply:
I had to run a little experiment of my own to be sure ... dividing by zero here indeed does result in +Inf instead of NaN or a trap or something.

One more thing: the rest of the argument checking is done in `NSQD.New()` (in `nsqd/nsqd.go`) so this should probably go there.

Reply:
Done.